File: DVLD_Project/Form6.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using BussnessLayerForDVLD;

namespace DVLD_Project
{
    public partial class Form6 : Form
    {
        public Form6()
        {
            InitializeComponent();
        }

        private void Form6_Load(object sender, EventArgs e)
        {
            DataTable dt = clsBussnesLayer.GetAllUsersShowing();
            guna2DataGridView1.DataSource = dt;
            cbFilters.Items.Add("None");
            guna2TextBox1.Visible = false;
            cbIsActive.Visible = false;
            cbFilters.SelectedIndex = 0;
            foreach (DataColumn column in dt.Columns)
            {
                cbFilters.Items.Add(column.ColumnName);
            }
        }

        private void guna2Button1_Click(object sender, EventArgs e)
        {
            Form7 form7 = new Form7();
            form7.ShowDialog();
        }

        private void cbFilters_SelectedIndexChanged(object sender, EventArgs e)
        {
            if (cbFilters.Text == "None")
            {
                guna2TextBox1.Visible = false;
                cbIsActive.Visible = false;
                guna2DataGridView1.DataSource = clsBussnesLayer.GetAllUsersShowing();
            }
            else if (cbFilters.Text != "IsActive")
            {
                guna2TextBox1.Visible = true;
                guna2DataGridView1.DataSource = clsBussnesLayer.GetAllUsersShowing();
                guna2TextBox1.Clear();
                cbIsActive.Visible = false;
            }
            else
            {
                guna2TextBox1.Visible = false;
                guna2DataGridView1.DataSource = clsBussnesLayer.GetAllUsersShowing();
                cbIsActive.Visible = true;
            }
        }

        private void guna2TextBox1_TextChanged(object sender, EventArgs e)
        {
            if (cbFilters.Text == "None" || cbFilters.Text == "")
            {

                guna2DataGridView1.DataSource = clsBussnesLayer.GetAllUsersShowing();
            }
            else if (cbFilters.Text == "PersonID" || cbFilters.Text == "UserID")
            {

                int Num = 0;
                if (guna2TextBox1.Text != "")
                {
                    int.TryParse(guna2TextBox1.Text, out Num);
                    if (Num == 0)
                    {
                        guna2TextBox1.Text = "";
                    }
                    else
                    {
                        guna2DataGridView1.DataSource = clsBussnesLayer.FindWithFiltersU(Num, cbFilters.Text, "");
                    }
                }
            }

            else
            {
                guna2DataGridView1.DataSource = clsBussnesLayer.FindWithFiltersU(0, cbFilters.Text, guna2TextBox1.Text);
            }
        }

        private void cbIsActive_SelectedIndexChanged(object sender, EventArgs e)
        {
            if (cbIsActive.Text != "None")
                guna2DataGridView1.DataSource = clsBussnesLayer.FindWithFiltersU(0, cbIsActive.Text, "");
            else
                guna2DataGridView1.DataSource = clsBussnesLayer.GetAllUsersShowing();
        }

        private void toolStripMenuItem4_Click(object sender, EventArgs e)
        {

        }

        private void toolStripMenuItem1_Click(object sender, EventArgs e)
        {
            Form8 form9 = new Form8();
            form9.ShowDialog();
        }

        private void toolStripMenuItem3_Click(object sender, EventArgs e)
        {
            if (clsBussnesLayer.DeleteUser(Convert.ToInt32(guna2DataGridView1.CurrentRow.Cells[0].Value)) == 0)
            {
                MessageBox.Show("YOY CANT");
            }
          
           
        }

        private void changePassWordToolStripMenuItem_Click(object sender, EventArgs e)
        {
            Form9 form9 = new Form9(Convert.ToInt32(guna2DataGridView1.CurrentRow.Cells[0].Value));  
            form9.ShowDialog();
        }

        private void toolStripMenuItem2_Click(object sender, EventArgs e)
        {
            Form7 form7 = new Form7(Convert.ToInt32(guna2DataGridView1.CurrentRow.Cells[1].Value), Convert.ToInt32(guna2DataGridView1.CurrentRow.Cells[0].Value));
            form7.ShowDialog();
        }
    }
}

File: DVLD_Project/Form6.vb
Imports System
Imports System.Data
Imports System.Windows.Forms
Imports BussnessLayerForDVLD

Namespace DVLD_Project
    Public Partial Class Form6
        Inherits Form
        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub Form6_Load(sender As Object, e As EventArgs)
            Dim dt As DataTable = clsBussnesLayer.GetAllUsersShowing()
            guna2DataGridView1.DataSource = dt
            cbFilters.Items.Add("None")
            guna2TextBox1.Visible = False
            cbIsActive.Visible = False
            cbFilters.SelectedIndex = 0
            For Each column As DataColumn In dt.Columns
                cbFilters.Items.Add(column.ColumnName)
            Next
        End Sub

        Private Sub guna2Button1_Click(sender As Object, e As EventArgs)
            Dim form7 As Form7 = New Form7()
            form7.ShowDialog()
        End Sub

        Private Sub cbFilters_SelectedIndexChanged(sender As Object, e As EventArgs)
            If Equals(cbFilters.Text, "None") Then
                guna2TextBox1.Visible = False
                cbIsActive.Visible = False
                guna2DataGridView1.DataSource = clsBussnesLayer.GetAllUsersShowing()
            ElseIf Not Equals(cbFilters.Text, "IsActive") Then
                guna2TextBox1.Visible = True
                guna2DataGridView1.DataSource = clsBussnesLayer.GetAllUsersShowing()
                guna2TextBox1.Clear()
                cbIsActive.Visible = False
            Else
                guna2TextBox1.Visible = False
                guna2DataGridView1.DataSource = clsBussnesLayer.GetAllUsersShowing()
                cbIsActive.Visible = True
            End If
        End Sub

        Private Sub guna2TextBox1_TextChanged(sender As Object, e As EventArgs)
            If Equals(cbFilters.Text, "None") OrElse Equals(cbFilters.Text, "") Then

                guna2DataGridView1.DataSource = clsBussnesLayer.GetAllUsersShowing()
            ElseIf Equals(cbFilters.Text, "PersonID") OrElse Equals(cbFilters.Text, "UserID") Then

                Dim Num = 0
                If Not Equals(guna2TextBox1.Text, "") Then
                    Integer.TryParse(guna2TextBox1.Text, Num)
                    If Num = 0 Then
                        guna2TextBox1.Text = ""
                    Else
                        guna2DataGridView1.DataSource = clsBussnesLayer.FindWithFiltersU(Num, cbFilters.Text, "")
                    End If
                End If
            Else
                guna2DataGridView1.DataSource = clsBussnesLayer.FindWithFiltersU(0, cbFilters.Text, guna2TextBox1.Text)
            End If
        End Sub

        Private Sub cbIsActive_SelectedIndexChanged(sender As Object, e As EventArgs)
            If Not Equals(cbIsActive.Text, "None") Then
                guna2DataGridView1.DataSource = clsBussnesLayer.FindWithFiltersU(0, cbIsActive.Text, "")
            Else
                guna2DataGridView1.DataSource = clsBussnesLayer.GetAllUsersShowing()
            End If
        End Sub

        Private Sub toolStripMenuItem4_Click(sender As Object, e As EventArgs)

        End Sub

        Private Sub toolStripMenuItem1_Click(sender As Object, e As EventArgs)
            Dim form9 As Form8 = New Form8()
            form9.ShowDialog()
        End Sub

        Private Sub toolStripMenuItem3_Click(sender As Object, e As EventArgs)
            If clsBussnesLayer.DeleteUser(Convert.ToInt32(guna2DataGridView1.CurrentRow.Cells(0).Value)) = 0 Then
                MessageBox.Show("YOY CANT")
            End If


        End Sub

        Private Sub changePassWordToolStripMenuItem_Click(sender As Object, e As EventArgs)
            Dim form9 As Form9 = New Form9(Convert.ToInt32(guna2DataGridView1.CurrentRow.Cells(0).Value))
            form9.ShowDialog()
        End Sub

        Private Sub toolStripMenuItem2_Click(sender As Object, e As EventArgs)
            Dim form7 As Form7 = New Form7(Convert.ToInt32(guna2DataGridView1.CurrentRow.Cells(1).Value), Convert.ToInt32(guna2DataGridView1.CurrentRow.Cells(0).Value))
            form7.ShowDialog()
        End Sub
    End Class
End Namespace
